client/src/API/API.ts

```diff
@@ -28,9 +30,265 @@ const logout = async () => {
+      stakeholder: "",
+      scale: "",
+      issuanceDate: null,
+      type: "",
+      connections: "",
+      language: "",
+      pages: null,
+      coordinates: { latitude: 67.8558, longitude: 20.2265 },
+    },
+    {
+      id: 5,
+      title: "Adjusted development plan (47)",
+      description:
+        "This document is the update of the Development Plan, one year after its creation, modifications are made to the general master plan, which is published under the name 'Adjusted Development Plan91,' and still represents the version used today after 10 years. Certainly, there are no drastic differences compared to the previous plan, but upon careful comparison, several modified elements stand out. For example, the central square now takes its final shape, as well as the large school complex just north of it, which appears for the first time.",
+      stakeholder: "",
+      scale: "",
+      issuanceDate: null,
+      type: "",
+      connections: "",
+      language: "",
+      pages: null,
+      coordinates: { latitude: 67.856, longitude: 20.225 },
+    },
+    {
+      id: 6,
+      title: "Document 6",
+      description: null,
+      stakeholder: "",
+      scale: "",
+      issuanceDate: null,
+      type: "",
+      connections: "",
+      language: "",
+      pages: null,
+      coordinates: { latitude: 67.857, longitude: 20.228 },
+    },
+    {
+      id: 7,
+      title: "Document 7",
+      description: "Description 7",
+      stakeholder: "",
+      scale: "",
+      issuanceDate: null,
+      type: "",
+      connections: "",
+      language: "",
+      pages: null,
+      coordinates: { latitude: 67.858, longitude: 20.229 },
+    },
+    {
+      id: 8,
+      title: "Document 8",
+      description: "Description 8",
+      stakeholder: "",
+      scale: "",
+      issuanceDate: null,
+      type: "",
+      connections: "",
+      language: "",
+      pages: null,
+      coordinates: { latitude: 67.859, longitude: 20.23 },
+    },
+    {
+      id: 9,
+      title: "Document 9",
+      description: "Description 9",
+      stakeholder: "",
+      scale: "",
+      issuanceDate: null,
+      type: "",
+      connections: "",
+      language: "",
+      pages: null,
+      coordinates: { latitude: 67.86, longitude: 20.231 },
+    },
+  ];
+}
+
+async function postDocument(id: number, title: string, description: string) {
+  try {
+    const response = await fetch(baseURL + `/documents/${id}`, {
```

Comment:
post document shouldn't take an id since the id is assigned by the database. The url needs changing, the args of this method need to be edited.

Reply:
async function postDocument(document: Document){
    try{
       const response = await fetch(baseURL + `/documents`, {...}
     }
}
Document will be something like this: 
export interface Link {
  targetDocumentId: number;
  type: string[];
}

export interface Document {
  id: number | string;
  title: string;
  description: string | null;
  stakeholder: string | undefined;
  scale: string | undefined;
  issuanceDate: Date | null;
  type: string | undefined;
  connections: Link[] | undefined;
  language: string | undefined;
  pages: number | null;
  coordinates: { latitude: number | null; longitude: number | null };
} 
and in the body i will pass all the parameters.

Reply:
document id is a number. Also description is mandatory. We need a separate discussion on connections